tmk_core/avr.mk

```diff
@@ -245,6 +245,14 @@ avrdude-split-left: $(BUILD_DIR)/$(TARGET).hex check-size cpfirmware
 avrdude-split-right: $(BUILD_DIR)/$(TARGET).hex check-size cpfirmware
 	$(call EXEC_AVRDUDE,eeprom-righthand.eep)
 
+BOOTLOADHID_PROGRAMMER ?= bootloadHID
+bootloadHID: $(BUILD_DIR)/$(TARGET).hex check-size cpfirmware
+	until lsusb -d 16c0:05df; do\
```

Comment:
This only works on linux, right?

Reply:
I figured it would be okay given that `tmk_core/avr.mk` was doing the following for dfu...

```
ls /dev/tty* > /tmp/1; \
```

While I do not have a MSYS2 environment at the moment, i would be happy to build one if it helps progress this PR.

Reply:
That's the AVRDUDE target, actually
And that does work, as it's listing the devices. 

However, lsusb doesn't appear to exist in MSYS2, at all. And I'm not really finding anything to go over how to install it.  

WSL seems to include it, though.

Reply:
My mistake.
I am not finding a detection method that will work on all supported platforms. I see a few methods forward.

1. Linux/macOS - continue with `lsusb`
macOS requires some magic as `lsusb` is not a default package
`brew install lsusb`
`brew install mikhailai/misc/usbutils`
1. Linux/macOS - inspect `/sys/bus/hid/devices/`
1. ALL - single sleep then blind flash attempt
Not the most reactive solution as the wait needs to be long enough to allow time for reboot but not long enough to annoy the user

Reply:
If there isn't one for all OS's, that's fine.  

But then we do need to ensure that there are methods that work for the other systems too, ideally. 

Windows being the biggest PITA here.

Reply:
I can detail any non compatible platform (including alt methods) within the work item "Convert existing bootloadHID docs centralised version". At the moment, the vast majority of readmes suggest the GUI tool for windows, so i would assume this is already common practice.

Maybe @mechmerlin has some insight with his recent BMC ports?

Reply:
One additional cross platform method could be to just keep running bootloadHid if the output contains "The specified device was not found"